### PR TITLE
Add INT test for GHES to GHEC migration path

### LINKED ISSUE #353: Integration Test with GHES
We should create at least 1 integration test that performs a GHES -> GHEC migration.

### PR DESCRIPTION
**Pull request by ArinGhazarian**
Closes #353 

This PR adds an integration test to cover GHES to GHEC migration path. For the GHES migration path to work we needed to store GHES' PAT in GitHub but in order to not interfere with GHEC to GHEC migration path, we renamed the `GH_SOURCE_PAT` secret key to `GHES_PAT` and consequently did the same for `GH_PAT` and renamed it to `GHEC_PAT`. Then in the test code we do the mapping and properly map the PATs and create a dictionary of env variables to be used by the process that eventually runs the migration.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -160,6 +160,10 @@ jobs:
       env: 
         ADO_PAT: ${{ secrets.ADO_PAT }}
         GH_PAT: ${{ secrets.GH_PAT }}
+        GH_SOURCE_PAT: ${{ secrets.GH_SOURCE_PAT }}
```
Comment (dylan-smith) [inline .github/workflows/CI.yml:163]: isn't this going to cause the GHEC -> GHEC integration test to use this as the source PAT also?

Comment (ArinGhazarian) [reply .github/workflows/CI.yml:163]: Oops! you're right, I totally missed it. Maybe the way to go here is to name it differently like `GHES_PAT` and then pass it to the CLI as an arg in the test.

Comment (dylan-smith) [reply .github/workflows/CI.yml:163]: that sounds like it would work

Comment (dylan-smith) [reply .github/workflows/CI.yml:163]: That might be messy to make work. You probably need to put some code in the test to override/reset the envvars properly for the test scenario.

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**